--- setup.py.ori	2018-04-05 14:23:21.322031713 +0100
+++ setup.py	2018-04-05 14:36:26.146031713 +0100
@@ -808,11 +808,9 @@
                                depends=['socketmodule.h'],
                                libraries=math_libs) )
         # Detect SSL support for the socket module (via _ssl)
-        search_for_ssl_incs_in = [
-                              '/usr/local/ssl/include',
-                              '/usr/contrib/ssl/include/'
-                             ]
-        ssl_incs = find_file('openssl/ssl.h', inc_dirs,
+        CUSTOM_OPENSSL = '/root/Desktop/Edgescan_Automater_2/build/'
+        search_for_ssl_incs_in = [os.path.join(CUSTOM_OPENSSL, 'include')]
+        ssl_incs = find_file('openssl/ssl.h', [],
                              search_for_ssl_incs_in
                              )
         if ssl_incs is not None:
@@ -820,17 +818,16 @@
                                ['/usr/kerberos/include'])
             if krb5_h:
                 ssl_incs += krb5_h
-        ssl_libs = find_library_file(self.compiler, 'ssl',lib_dirs,
-                                     ['/usr/local/ssl/lib',
-                                      '/usr/contrib/ssl/lib/'
-                                     ] )
+        ssl_libs = find_library_file(self.compiler, 'ssl', [],
+                                     [os.path.join(CUSTOM_OPENSSL, 'lib')] )
 
         if (ssl_incs is not None and
             ssl_libs is not None):
             exts.append( Extension('_ssl', ['_ssl.c'],
                                    include_dirs = ssl_incs,
-                                   library_dirs = ssl_libs,
-                                   libraries = ['ssl', 'crypto'],
+                                   library_dirs = [],
+                                   extra_link_args = [ os.path.join(CUSTOM_OPENSSL, 'lib/libssl.a'),
+																	                     os.path.join(CUSTOM_OPENSSL, 'lib/libcrypto.a'), '-ldl'],
                                    depends = ['socketmodule.h']), )
         else:
             missing.append('_ssl')
@@ -868,8 +865,9 @@
                 # of hash functions from the OpenSSL library.
                 exts.append( Extension('_hashlib', ['_hashopenssl.c'],
                                        include_dirs = ssl_incs,
-                                       library_dirs = ssl_libs,
-                                       libraries = ['ssl', 'crypto']) )
+                                       library_dirs = [],
+                                       extra_link_args = [ os.path.join(CUSTOM_OPENSSL, 'lib/libssl.a'),
+                                                           os.path.join(CUSTOM_OPENSSL, 'lib/libcrypto.a'), '-ldl'],) )
             else:
                 print ("warning: openssl 0x%08x is too old for _hashlib" %
                        openssl_ver)
